app/assets/javascripts/mirador_widget_admin.js

```diff
@@ -43,6 +49,35 @@
       });
     }
 
+    function modalMiradorSubmitListener(block) {
+      block.on('mirador-modal-closed', function(e, value) {
+        // Extract and merge config information
+        var miradorInstance = $(value.currentTarget).parents()
+          .find('iframe')[0].contentWindow.miradorInstance;
+        var config = miradorInstance.saveController.currentConfig;
+        var newConfig = {
+          data: config.data,
+          layout: config.layout,
+          windowObjects: config.windowObjects.map(function(value) {
+            return {
+              slotAddress: value.slotAddress,
+              viewType: value.viewType,
+              canvasID: value.canvasID,
+              loadedManifest: value.loadedManifest,
+              sidePanelVisible: value.sidePanelVisible,
+              windowOptions: value.windowOptions ? { osdBounds: value.windowOptions.osdBounds } : undefined
+            }
+          })
+        };
+        // Add config to hidden form.
+        var input = '<input type="text" style="display:none;" name="mirador_config" value=\'' +
+         JSON.stringify(newConfig) +
```

Comment:
I'm immediately suspicious of this and wonder if there are any data escaping concerns we should be concerned about?

Reply:
Should be fixed now in bb681c9864227114dd10456d55833cbc4ba035f5 . Thanks for the suggestion!